### PR TITLE
#261: Invalid plan when re-applying terraform module

### LINKED ISSUE #261: Invalid plan when re-applying terraform module
When creating a gke cluster using the beta-private-cluster module, I'm running into issues when I apply the configuration again. It works fine for the first apply, subsequent plans and applies fail for the same code. 

Terraform version
```
Terraform v0.12.8
+ provider.google v2.14.0
+ provider.google-beta v2.14.0
+ provider.kubernetes v1.9.0
+ provider.null v2.1.2
+ provider.random v2.2.0
```

Getting the following error
```
------------------------------------------------------------------------

Error: Provider produced invalid plan

Provider "google-beta" has indicated "requires replacement" on
module.gke.google_container_cluster.primary for a non-existent attribute path
cty.Path{cty.GetAttrStep{Name:"private_cluster_config"},
cty.IndexStep{Key:cty.NumberIntVal(0)},
cty.GetAttrStep{Name:"master_ipv4_cidr_block"}}.
```

The module looks something like this

```
module "gke" {

  source = "git::https://github.com/terraform-google-modules/terraform-google-kubernetes-engine.git//modules/beta-private-cluster?ref=master"

  name                       = local.kubernetes_cluster_name
  project_id                 = "${var.project_name}"
  region                     = "${var.region}"
  network                    = "${data.terraform_remote_state.network.outputs.network_name}"
  subnetwork                 = data.terraform_remote_state.network.outputs.subnets_names[0]
  ip_range_pods              = "${data.terraform_remote_state.network.outputs.subnets_names[0]}-1"
  ip_range_services          = "${data.terraform_remote_state.network.outputs.subnets_names[0]}-2"
  http_load_balancing        = "${var.http_load_balancing}"
  horizontal_pod_autoscaling = "${var.horizontal_pod_autoscaling}"
  kubernetes_dashboard       = "${var.kubernetes_dashboard}"
  network_policy             = "${var.network_policy}"
  monitoring_service         = "${var.kubernetes_monitoring_service}"
  logging_service            = "${var.kubernetes_logging_service}"
  create_service_account     = false
  service_account            = "${var.service_account}"
  enable_private_endpoint    = "${var.enable_private_endpoint}"
  enable_private_nodes       = "${var.enable_private_nodes}"
  remove_default_node_pool   = "${var.remove_default_node_pool}"

  master_authorized_networks_config = [
    {
      cidr_blocks = [
        {
          cidr_block   = "xx.xx.xx.xx/32"
          display_name = "Whitelist"
        },
      ]
    },
  ]

  node_pools = [
    {
      name               = "default-node-pool"
      machine_type       = "${var.kubernetes_machine_type}"
      disk_size_gb       = "${var.kubernetes_node_pool_disk_size}"
      disk_type          = "${var.kubernetes_node_pool_disk_type}"
      image_type         = "${var.kubernetes_node_pool_image_type}"
      auto_repair        = true
      auto_upgrade       = true
      service_account    = "${var.service_account}"
      preemptible        = false
      initial_node_count = "${var.initial_node_count}"
    },
  ]

  node_pools_oauth_scopes = {
    all = []

    default-node-pool = [
      "https://www.googleapis.com/auth/cloud-platform",
    ]
  }
}
```

Using master because a PR that was causing issues for me was merged and a new release hasn't been tagged. Please let me know if I can provide more info

### PR DESCRIPTION
**Pull request by nick4fake**
Fixes #261

Bumbed minimal provider version to 2.18

### CODE REVIEW

File: autogen/versions.tf [hunk 1]
```diff
@@ -17,3 +17,13 @@
 terraform {
   required_version = ">= 0.12"
 }
+
+{% if beta_cluster %}
+provider "google-beta" {
```
Comment (morgante) [inline autogen/versions.tf]: We should not specify a version in the module itself.

Instead, we should simply specify [required versions](https://github.com/forseti-security/terraform-google-forseti/blob/master/versions.tf#L20).

File: examples/disable_client_cert/main.tf [hunk 2]
```diff
@@ -19,8 +19,7 @@ locals {
 }
 
 provider "google" {
-  version = "~> 2.12.0"
```
Comment (morgante) [inline examples/disable_client_cert/main.tf]: Don't remove the version pinning from examples.

**Comment by rbramwell** [comment]
@nick4fake @morgante This is preventing me from using the module with the 2.20.0 google provider version. As per https://www.terraform.io/docs/configuration/terraform.html#specifying-required-provider-versions greater than or equal >= would have been a better option instead of the pessimistic constraint ~>.

**Comment by rbramwell** [comment]
Saw the CHANGELOG.md. Looks like I won't be able to use this with anything but ~> 2.18.0 google provider version. I will resort to continue using the 5.1.1 version of the module.

**Comment by morgante** [comment]
@rbramwell Sorry for the issue, we've relaxed the version constraints with #359. If you upgrade to 6.0.1 you should be able to use 2.20.0.

**Pull request merged**